// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package privacy_sandbox.server_common.byob;

message DispatcherRequest {
  message LoadBinary {
    bytes code_token = 1;
    // Content to load request can be provided one of two ways. Namely -
    // 1. binary_content: Contents of the pre-complied, self-contained binary
    // 2. source_bin_code_token: Code token of a existing binary in the sandbox.
    // For instance, an existing binary might be reloaded in for logging.
    oneof content_oneof {
      bytes binary_content = 2;
      bytes source_bin_code_token = 5;
    }
    int32 num_workers = 3;
    // Whether or not logs should be stored in a file. Set to false by proto3
    // default.
    bool enable_log_egress = 4;
  }

  message DeleteBinary {
    bytes code_token = 1;
  }
  message Cancel {
    bytes execution_token = 1;
  }

  oneof request {
    LoadBinary load_binary = 1;
    DeleteBinary delete_binary = 2;
    Cancel cancel = 3;
  }
}
